packages/firestore/src/local/mutation_queue.ts

```diff
@@ -81,14 +81,22 @@ export interface MutationQueue extends GarbageSource {
     streamToken: ProtoByteString
   ): PersistencePromise<void>;
 
-  /** Creates a new mutation batch and adds it to this mutation queue. */
+  /**
+   * Creates a new mutation batch and adds it to this mutation queue.
+   *
+   * TODO(Multi-Tab): Make this operation safe to use from secondary clients.
+   */
   addMutationBatch(
     transaction: PersistenceTransaction,
     localWriteTime: Timestamp,
     mutations: Mutation[]
   ): PersistencePromise<MutationBatch>;
 
-  /** Loads the mutation batch with the given batchId. */
+  /**
+   * Loads the mutation batch with the given batchId.
+   *
+   * Multi-Tab Note: This operation is safe to use from secondary clients.
```

Comment:
FWIW I wasn't imagining plumbing it all the way down to SimpleDbStore since I figured it'd be too context-sensitive (e.g. secondary tabs can add mutation queue objects, but only the primary can delete them).  So I was imagining they'd live in the MutationQueue / RemoteDocumentCache / etc. classes.  Classes that need the primary lease would just assert transaction.hasPrimaryLease() or something.

But I don't feel very strongly about doing this. It may be more noise than it's worth. So don't bother unless you are more convinced than me.